### PR TITLE
[RFR] Convert MenuLinkItem component to useStyles

### PR DESCRIPTION
**Pull request by jaytula**
Convert MenuLinkItem component to useStyles. #3453

### CODE REVIEW

File: packages/ra-ui-materialui/src/layout/MenuItemLink.js [hunk 1]
```diff
@@ -1,61 +1,43 @@
...
-        classes: PropTypes.object.isRequired,
-        className: PropTypes.string,
-        leftIcon: PropTypes.element,
-        onClick: PropTypes.func,
-        primaryText: PropTypes.node,
-        staticContext: PropTypes.object,
-        to: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
-            .isRequired,
-    };
+export function MenuItemLink({ classes: classesOverride, ...props }) {
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: I think it isn't needed to export this anymore.

File: packages/ra-ui-materialui/src/layout/MenuItemLink.js [hunk 2]
```diff
@@ -1,61 +1,43 @@
...
-            classes,
-            className,
-            primaryText,
-            leftIcon,
-            staticContext,
-            sidebarIsOpen,
-            ...props
-        } = this.props;
+    const renderMenuItem = () => {
+        const { className, primaryText, leftIcon } = props;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: Extract `className`, `primaryText` and `leftIcon` from the props at L24

File: packages/ra-ui-materialui/src/layout/MenuItemLink.js [hunk 3]
```diff
@@ -1,61 +1,43 @@
...
-        to: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
-            .isRequired,
-    };
+export function MenuItemLink({ classes: classesOverride, ...props }) {
+    const classes = useStyles({ classes: classesOverride });
 
-    handleMenuTap = e => {
-        this.props.onClick && this.props.onClick(e);
+    const handleMenuTap = e => {
+        props.onClick && props.onClick(e);
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: Extract this from the props at L24

Comment (djhi) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: Extract `onClick` from the props at L24

File: packages/ra-ui-materialui/src/layout/MenuItemLink.js [hunk 4]
```diff
@@ -65,21 +47,29 @@ export class MenuItemLink extends Component {
...
-    }
-
-    render() {
-        const { sidebarIsOpen, primaryText } = this.props;
+    };
 
-        if (sidebarIsOpen) {
-            return this.renderMenuItem();
-        }
+    const { sidebarIsOpen, primaryText } = props;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: Same extract those props at L24

File: packages/ra-ui-materialui/src/layout/MenuItemLink.js [hunk 5]
```diff
@@ -1,61 +1,49 @@
...
+    leftIcon,
+    onClick,
+    sidebarIsOpen,
+    ...props
+}) {
+    const classes = useStyles({ classes: classesOverride });
 
-    handleMenuTap = e => {
-        this.props.onClick && this.props.onClick(e);
+    const handleMenuTap = e => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/MenuItemLink.js]: this one should be memoized with `useCallback`

Comment (jaytula) [reply packages/ra-ui-materialui/src/layout/MenuItemLink.js]: Sure, I've done this. Thanks for looking at it!

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**